app/server.js

```diff
@@ -31,39 +31,83 @@ for (const [loggerName, logLevel] of Object.entries(config.loglevel)) {
 
 log.info(`==== Open Rowing Monitor ${process.env.npm_package_version || ''} ====\n`)
 
+if (config.gpioPriority) {
+  // setting a higher (near-real-time?) priority for the main process, as we don't want to block the GPIO thread
+  const mainPriority = Math.min((config.gpioPriority + 2), 0)
```

Comment:
What is the scenario where we need to alter the the Nice-Setting of the main application as applied here? I can totally see the situation where we want higher priority on the TimerService, as that one needs to measure time as precise as possible 
However the rest of the application does not have such a demand, so I'm not sure what the benefit of this is. If we draw resources from the system like this I'm a bit worried that this might cause other system things like Browser, Network and Scheduling to suffer from this.
If there is a use case for this, then I would prefer, that we define this by setting two variables in `default.config.js`, like `gpioPriority` and `appPriority` as coupling them with some 'magic formula' like this can make tweaking kind of confusing.

Reply:
Good point!

The scenario is that when you run on a high frequency machine on a high priority for the gpio, the app priority might be too low to consume the data. I noticed strange behaviour with the `onoff` implementation, as `GpioTimerService.js` seemed it couldn't deliver its data to `server.js`. This was resolved by increasing the priority of the main proces, allowing it to respond quickly.

With the new `pigpio` I kept it in place. I don't know if it is still required as `pigpio` has its own internal timer and queueing. However, I do notice that at nice level 0 you are frequently fighting for CPU time with all non-time critical processes (like many housekeeping processes in Linux like the time and FIPS updates which can't be controlled).

I recognize that setting the nice level too high could kill Network, Scheduling and above all Timekkeeping. This happens when your Nice-level gets too high (as documented in `defualt.config.js`). Browser should be a lower priority: on a C2 the app is updated every 9 ms, and the browser is updated every 80 ms. Updating it a couple of miliseconds later wouldn't be visible to the user.

I'll update the app with your suggesting as it is a good way to improve it. These magic tricks are indeed confusing, and we should avid that.

BTW: thanks for updating the naming for the VO2Max function. I was a bit busy with fixing an issue with my flywheel, and I forgot about this open issue.